core-bundle/contao/drivers/DC_Folder.php

```diff
@@ -2912,62 +2896,66 @@ protected function isMounted($strFolder)
 	/**
 	 * Check a file operation
 	 *
-	 * @param string $strFile
+	 * @param string $strFileOrFolder
 	 *
 	 * @return boolean
 	 *
 	 * @throws AccessDeniedException
 	 */
-	protected function isValid($strFile)
+	protected function isValid($strFileOrFolder)
 	{
-		$strFolder = Input::get('pid', true);
-
 		// Check the path
-		if (Validator::isInsecurePath($strFile))
+		if (Validator::isInsecurePath($strFileOrFolder))
 		{
-			throw new AccessDeniedException('Invalid file name "' . $strFile . '" (hacking attempt).');
+			throw new AccessDeniedException('Invalid file or folder name "' . $strFileOrFolder . '" (hacking attempt).');
 		}
 
-		if ($strFolder && Validator::isInsecurePath($strFolder))
-		{
-			throw new AccessDeniedException('Invalid folder name "' . $strFolder . '" (hacking attempt).');
-		}
```

Comment:
We must not remove the parent folder check, because existing code relies on it being executed. For example:

https://github.com/contao/contao/blob/41ddcc7784e3c0ebeefa0c469c9eba3a63deca4c/core-bundle/contao/drivers/DC_Folder.php#L675-L685

If we remove the parent folder check, we would have to add a `$this->isValid($strFolder);` before line 682.